vortex-expr/src/identity.rs

```diff
@@ -22,7 +22,7 @@ impl PartialEq<dyn Any> for Identity {
     fn eq(&self, other: &dyn Any) -> bool {
         unbox_any(other)
             .downcast_ref::<Self>()
-            .map(|x| x == other)
+            .map(|x| x == self)
```

Comment:
lol this is a real bug

Reply:
Yep, I can split it from this pr but until this pr this expression has no usages